p8e-api/src/main/kotlin/io/provenance/engine/batch/MailboxReaper.kt

```diff
@@ -240,7 +241,10 @@ class MailboxReaper(
                 }
             }.mapNotNull { (keyPair, result) ->
                 val (uuid, dimeInputStream) = result
-                dimeInputStream.getDecryptedPayload(keyPair).use {
+                val signer = transaction{ affiliateService.getSigner(keyPair.public) }
+                val encryptionKeyRef = transaction{ affiliateService.getEncryptionKeyRef(keyPair.public) }
+
+                dimeInputStream.getDecryptedPayload(encryptionKeyRef, signer).use {
```

Comment:
Does this work in all cases? I think the one parameter function `getDecryptedPayload(keyPair)` would use the first signer to verify. Passing an explicit signer would mean it fails verification if that signer didn't sign? Wouldn't that be the case for mailbox items because it's the sender that signed?

Reply:
The purpose of the signer for this particular call is to tell P8e which signer to init, actual verification doesn't happen until a little later below.